src/broadcast/scrobblingmanager.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <QObject>
+#include <QMutex>
+#include <QLinkedList>
+
+#include "broadcast/metadatabroadcast.h"
+#include "mixer/basetrackplayer.h"
+#include "track/track.h"
+#include "track/tracktiminginfo.h"
+
+class BaseTrackPlayer;
+class PlayerManager;
+
+class ScrobblingManager : public QObject {
```

Comment:
I think the name of this class is not self explaining. 
How many "Current tracks" we will have finally? 
* The current scrobble track 
* The current Track for Metadata Streaming (RDS) 
* The current Track for history playlist 
* The current Track driving the master clock for beat matching (@kshitij98 project) 

Will this become a CurrentTrackManager or CurrentTrackCalculator? 

@kshitij98: could you reuse portions of this code? Or could you share your rules for a master clock track?

Reply:
The current scrobble track, metadata streaming and history playlist are all the same, the one given by PlayerInfo.h. This manager takes into account when a track is loaded, paused, etc. and checks if they're audible every second to update the timers, and when they are it sends a scrobble signal.

Reply:
> @kshitij98: could you reuse portions of this code? Or could you share your rules for a master clock track?

`PlayerInfo::getCurrentPlayingTrack` returns the most audible track which should drive the master clock. This is not a part of my project so I will be able to continue working on it only once I'm done with my project, or I have sufficient time in between.

Related: https://bugs.launchpad.net/mixxx/+bug/1770239